trustchain-trader/src/main/java/nl/tudelft/trustchain/trader/ui/TraderFragment.kt

```diff
@@ -3,29 +3,141 @@ package nl.tudelft.trustchain.trader.ui
+        binding.acceptedPayloads.addItemDecoration(DividerItemDecoration(context, LinearLayout.VERTICAL))
+
+        binding.declinedPayloads.adapter = adapterDeclined
+        binding.declinedPayloads.layoutManager = LinearLayoutManager(context)
+        binding.declinedPayloads.addItemDecoration(DividerItemDecoration(context, LinearLayout.VERTICAL))
+
+        val marketCommunity = getMarketCommunity()
+        marketCommunity.addListener(TradePayload.Type.ASK, ::askListener)
+        marketCommunity.addListener(TradePayload.Type.BID, ::bidListener)
+
+        switchTrader.setOnClickListener {
+            if (!isTrading) {
+                TrustChainTraderActivity.acceptedPayloads.add(
+                    TradePayload(
+                        trustchain.getMyPublicKey(),
+                        Currency.DYMBE_DOLLAR,
+                        Currency.BTC,
+                        43.0,
+                        13.0,
+                        TradePayload.Type.ASK
                     )
+                )
+            }
+            isTrading = !isTrading
+        }
+        loadCurrentPayloads((TrustChainTraderActivity.acceptedPayloads), "accepted")
+        loadCurrentPayloads((TrustChainTraderActivity.declinedPayloads), "declined")
+    }
+
+    private fun loadCurrentPayloads(
+        payloads: List<TradePayload>,
+        adapterString: String
+    ) {
+        lifecycleScope.launchWhenStarted {
+            while (isActive) {
+                val items = withContext(Dispatchers.IO) {
+                    payloads.map {
+                        PayloadItem(
+                            it.publicKey,
+                            it.primaryCurrency,
+                            it.secondaryCurrency,
+                            it.amount,
+                            it.price,
+                            it.type
+                        )
+                    }
                 }
-            })
+                if (adapterString == "accepted") {
+                    adapterAccepted.updateItems(items)
+                }else if (adapterString == "declined") {
+                    adapterDeclined.updateItems(items)
+                }
+
+                binding.imgEmpty.isVisible = items.isEmpty() && (TrustChainTraderActivity.acceptedPayloads).isEmpty() && (TrustChainTraderActivity.declinedPayloads).isEmpty()
+
+                delay(1000)
+            }
         }
     }
+
+    private fun askListener(payload: TradePayload) {
+        val ai = NaiveBayes(resources.openRawResource(R.raw.trustchain_trade_data_v7_int))
+        Log.d(
+            "PayloadFragment::onViewCreated",
+            "New ask came in! They are selling ${payload.amount} ${payload.primaryCurrency}. The price is ${payload.price} ${payload.secondaryCurrency} per ${payload.primaryCurrency}"
+        )
+        var type = if(payload.type == TradePayload.Type.ASK) 0 else 1
+        if (ai.predict(payload.amount!!.roundToInt() / payload.price!!.roundToInt(), type) == 1){
```

Comment:
The input to the predict function should be the price in Dymbe $ for 1 btc. I'm not sure this works.